plugin/src/main/java/com/spotify/plugin/dockerfile/BuildMojo.java

```diff
@@ -158,11 +164,28 @@ static String buildImage(@Nonnull DockerClient dockerClient,
     }
 
     if (buildArgs != null && !buildArgs.isEmpty()) {
-      try {
-        final String encodedBuildArgs = URLEncoder.encode(new Gson().toJson(buildArgs), "utf-8");
-        buildParameters.add(new DockerClient.BuildParam("buildargs", encodedBuildArgs));
-      } catch (UnsupportedEncodingException e) {
-        throw new MojoExecutionException("Could not build image", e);
+      buildParameters.add(new DockerClient.BuildParam("buildargs", encodeBuildParam(buildArgs)));
+    }
+
+    if (cacheFrom != null) {
+      final List<String> cacheFromExistLocally = new ArrayList<>();
+      for (String image : cacheFrom) {
+        try {
+          if (pullNewerImage || !imageExistLocally(dockerClient, image)) {
+            dockerClient.pull(image);
+          }
+          log.info(MessageFormat.format("Build will use image {0} for cache-from", image));
+          cacheFromExistLocally.add(image);
+        } catch (ImageNotFoundException e) {
+          log.warn(MessageFormat.format(
+                  "Image {0} not found, build will not use it for cache-from", image));
+        } catch (DockerException | InterruptedException e) {
+          throw new MojoExecutionException("Could not pull cache-from image", e);
+        }
+      }
+      if (!cacheFromExistLocally.isEmpty()) {
+        buildParameters.add(new DockerClient.BuildParam("cache-from",
```

Comment:
this should be "cachefrom" not "cache-from" 

https://docs.docker.com/engine/api/v1.40/#operation/ImageBuild